instrumentation/http_instrumentation.go

```diff
@@ -91,10 +102,67 @@ func (m instrumentedHandlerFactory) NewHandler(extraLabels prometheus.Labels, ne
 }
 
 // NewInstrumentedHandlerFactory creates a new instrumentedHandlerFactory.
-func NewInstrumentedHandlerFactory(req *prometheus.Registry, hardcodedLabels []string) instrumentedHandlerFactory {
+func NewInstrumentedHandlerFactory(req *prometheus.Registry) instrumentedHandlerFactory {
 	return instrumentedHandlerFactory{
-		metricsCollector: newHTTPMetricsCollector(req, hardcodedLabels),
+		metricsCollector: newHTTPMetricsCollector(req),
+	}
+}
+
+func httpHandlerLabels(r *http.Request) prometheus.Labels {
+	extraLabels := prometheus.Labels{
+		"handler": "unknown",
+		"group":   "unknown",
+	}
+
+	routePattern := chi.RouteContext(r.Context()).RoutePattern()
+	tenant, ok := authentication.GetTenant(r.Context())
+	if !ok {
+		return extraLabels
 	}
+
+	var groupHandler map[string]groupHandler
+	switch routePattern {
+	case "/api/metrics/v1/{tenant}/*":
+		groupHandler = metricsV1Group
+	case "/api/v1/{tenant}/*":
+		groupHandler = legacyMetricsGroup
+	}
+
+	strippedPath := strings.Split(r.URL.Path, tenant)
+	if len(strippedPath) != 2 {
+		return extraLabels
+	}
+
+	gh, ok := groupHandler[strippedPath[1]]
+	if ok {
+		extraLabels = prometheus.Labels{
+			"group":   gh.group,
+			"handler": gh.handler,
+		}
+	}
+	return extraLabels
+}
+
+type groupHandler struct {
+	group   string
+	handler string
+}
+
+var legacyMetricsGroup = map[string]groupHandler{
+	metricslegacy.QueryRoute:      {"metricslegacy", "query"},
```

Comment:
Is it necessary to move these here and import `metricslegacy` or `metricsv1` in `instrumentation`? I fear that at some point this would lead to cyclic dependencies between things, when someone tries to import instrumentation stuff in the metrics packages. 

Maybe we can just add these maps to `main.go` or even in the `metrics` package itself?

Reply:
I think we should solve this from a different perspective. First, adding yet more things in `main.go` is becoming unsustainable: it has around 1k lines already and becomes a bucket where we throw all kinds of logic. Second, this map is meant only for **instrumentation** purposes: the group and handler labels. So I think it makes total sense to move them here, together with the rest of the instrumentation code.

In the `api/metrics` package we have only handlers being set up.